### PR TITLE
Fix wrong sweeped address in logs

### LINKED ISSUE #306: Wrong sweeping address in logs
Logs:
```
[2021-12-05T17:46:42Z INFO  farcaster_node::syncerd::monero_syncer] Sweeping address 56DzEiq4m8u4CYViSD2YdSUXmHKLninnZKDwsmYsz1e1RsbYrF5qcTjZ7ey9f1hQ4wjJ1D4B8rNMZD7J9YH8zbjoVrwjcnK with unlocked balance: 0.030000000000 XMR
```
and
```
[2021-12-05T17:46:43Z INFO  farcaster_node::syncerd::monero_syncer] Sweeping address 56DzEiq4m8u4CYViSD2YdSUXmHKLninnZKDwsmYsz1e1RsbYrF5qcTjZ7ey9f1hQ4wjJ1D4B8rNMZD7J9YH8zbjoVrwjcnK with unlocked balance: 0.020000000000 XMR
```
The displayed address is not the sweeped one but the destination address.

### PR DESCRIPTION
**Pull request by h4sh3d**
Fix #306

Also fix corresponding address after extracting secrets from Buy and Refund (follow up of #302)

### CODE REVIEW

File: src/syncerd/monero_syncer.rs [hunk 1]
```diff
@@ -271,7 +271,7 @@ async fn sweep_address(
     if balance.unlocked_balance != 0 {
         info!(
             "Sweeping address {} with unlocked balance: {}",
-            dest_address.addr(),
```
Comment (TheCharlatan) [inline src/syncerd/monero_syncer.rs:274]: Why not both?

Comment (h4sh3d) [reply src/syncerd/monero_syncer.rs:274]: I'll add both. Just saw that there is another log error somewhere else... will fix it in this PR too

Comment (h4sh3d) [reply src/syncerd/monero_syncer.rs:274]: Done in 4a638ca0bc897ece090e9d0aa48359791a920be8

**Review by TheCharlatan — APPROVED** [review]
CI cache also seems to be working nicely now.

**Pull request merged**